### PR TITLE
Add Vertex AI RunInference Notebook

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds a runnable notebook example for the Vertex AI image classification example.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
R: @damccorm

File: examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "HrCtxslBGK8Z"
+      },
+      "source": [
+        "This notebook shows how to use the Apache Beam [RunInference](https://beam.apache.org/releases/pydoc/current/apache_beam.ml.inference.base.html#apache_beam.ml.inference.base.RunInference) transform for image classification with [Vertex AI](https://cloud.google.com/vertex-ai).\n",
+        "Apache Beam has built-in support for sending requests to a remotely deployed Vertex AI endpoint using the [`VertexAIModelHandlerJSON`](https://github.com/apache/beam/blob/395c4d15bb74351b0aa020dc7463de8d85766e07/sdks/python/apache_beam/ml/inference/vertex_ai_inference.py#L61) class. The input for this class is a JSON-serializable type, like a list.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb:56]:
```suggestion
        "Apache Beam has built-in support for sending requests to a remotely deployed Vertex AI endpoint by using the [`VertexAIModelHandlerJSON`](https://github.com/apache/beam/blob/395c4d15bb74351b0aa020dc7463de8d85766e07/sdks/python/apache_beam/ml/inference/vertex_ai_inference.py#L61) class. The input for this class is a JSON-serializable type, like a list.\n",
```

File: examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+        "\n",
+        "When you use remote inference with Vertex AI, consider the following factors:\n",
+        "1. Public endpoints have a maximum request size of 1.5 MB. If you want to send larger requests, you must configure a [private endpoint](https://cloud.google.com/vertex-ai/docs/predictions/using-private-endpoints) and run your pipeline within the same VPC network. You might want to send larger requests if you send batches of requests.\n",
+        "2. Inputs to the Vertex AI model handler must be JSON serializable. If the inputs that aren't JSON serializable, the request to the endpoint fails.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb:60]:
```suggestion
        "2. Inputs to the Vertex AI model handler must be JSON serializable. If the inputs aren't JSON serializable, the request to the endpoint fails.\n",
```

File: examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "YDHPlMjZRuY0"
+      },
+      "source": [
+        "### Install Apache Beam\n",
+        "To use RunInference with the built-in Vertex AI model handler, install Apache Beam version 2.50.0 or later.\n",
+        "\n",
+        "**Note:** at time of writing 2.50.0 is in the release process and is not available, so this notebook uses a release candidate build."
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb:100]:
```suggestion
        "**Note:** The Apache Beam 2.50.0 SDK is currently in the release process and is not available. This notebook uses a release candidate build."
```

File: examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "YDHPlMjZRuY0"
+      },
+      "source": [
+        "### Install Apache Beam\n",
+        "To use RunInference with the built-in Vertex AI model handler, install Apache Beam version 2.50.0 or later.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb:98]:
```suggestion
        "To use RunInference with the built-in Vertex AI model handler, install the Apache Beam SDK version 2.50.0 or later.\n",
```

File: examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+          },
+          "execution_count": 3,
+          "metadata": {},
+          "output_type": "execute_result"
+        }
+      ],
+      "source": [
+        "aiplatform.init(project=project, location=location)\n",
+        "endpoint = aiplatform.Endpoint(endpoint_name=endpoint_id)\n",
+        "# You can get more metadata if desired by removing [0].display_name\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb:194]:
```suggestion
        "# To get more metadata, remove [0].display_name\n",
```

File: examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Y3BC2aY8cIMI"
+      },
+      "source": [
+        "### Preprocess an example image\n",
+        "\n",
+        "Preprocess an input to match what your model expects. Use the following code to complete these steps:\n",
+        "\n",
+        "1. Test the model by using an image of sunflowers.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb:208]: Is the image still of sunflowers?

File: examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+        "2. Trim the image to a 128x128 pixel size to match the model's expectations.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb:209]:
```suggestion
        "2. Trim the image to 128x128 pixels to match the model's expectations.\n",
```

File: examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb (hunk 6 above)
Comment (jrmccluskey) [reply examples/notebooks/beam-ml/run_inference_vertex_ai.ipynb:208]: Yes

**Review by rszper — APPROVED** [review]
LGTM

**Review by damccorm — APPROVED** [review]
Thanks!

**Pull request merged**